### PR TITLE
[Workspace] Rewrite filter

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes:
[AB#198421](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/198421)
[AB#138718](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/138718)
[AB#204310](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/204310)
[AB#203845](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/203845)
[AB#187263](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/187263)
[AB#203813](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/203813)
[AB#203838](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/203838)

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/packages/Filter/Components/Utils/debounceFilterItemCheck.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
-import { FilterItem, FilterItemCheck } from '../../Types/FilterItem';
```
Comment (Gustav-Eikaas) [inline src/packages/Filter/Components/Utils/debounceFilterItemCheck.ts:1]: delete file

File: src/packages/Filter/Hooks/useCount.ts [hunk 2]
```diff
@@ -1,46 +1,46 @@
...
-                return currentCount;
-            }
-
-            if (isLoading) return oldCount;
-            if (currentCount === 0) {
-                setIsActive(false);
-            }
-
-            return currentCount;
-        });
```
Comment (Gustav-Eikaas) [inline src/packages/Filter/Hooks/useCount.ts:38]: delete

File: src/packages/Filter/Hooks/useFilterApi.ts [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+    function getFilterItemCountsForGroup(groupName: string): FilterItemCount[] {
+        const filterGroup = allFilterValues.current.find(({ name }) => name === groupName);
+        if (!filterGroup) return [];
+
+        return filterGroup.values.map(
+            (value): FilterItemCount => ({
+                name: value,
+                count: filteredData.current.filter((item) =>
+                    doesItemPassFilter(
```
Comment (Gustav-Eikaas) [inline src/packages/Filter/Hooks/useFilterApi.ts:127]: extract

File: src/packages/Filter/Types/PersistFilter.ts [hunk 4]
```diff
@@ -1,17 +1,17 @@
-import { FilterData } from './FilterItem';
+// import { FilterData } from './FilterItem';
 
-export interface FilterSave {
-    filterData: FilterData;
-    name: string;
-}
-/**Options to persist filter */
-export interface FilterPersistOptions {
-    getFilter(): FilterData | undefined;
```
Comment (Gustav-Eikaas) [inline src/packages/Filter/Types/PersistFilter.ts:9]: delete

File: src/packages/Filter/Types/index.ts [hunk 5]
```diff
@@ -1,2 +1,2 @@
-export * from './FilterItem';
-export * from './PersistFilter';
+export * from './filter';
+// export * from './PersistFilter';
```
Comment (Gustav-Eikaas) [inline src/packages/Filter/Types/index.ts:2]: remove

File: src/packages/Filter/functions/generateFilterValues.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                const singleValue = handlePossiblyEmptyString(value);
+                /** Append value and prevent duplicates */
+                filterGroup.values = [
+                    ...filterGroup.values.filter((oldValue) => oldValue !== singleValue),
+                    singleValue,
+                ];
+            }
+        })
+    );
+    console.log(filterGroups);
```
Comment (Gustav-Eikaas) [inline src/packages/Filter/functions/generateFilterValues.ts:47]: remove

File: src/packages/Filter/functions/generateFilterValues.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+function sortFilterGroups<T = unknown>(
+    groups: FilterGroup[],
+    filterConfig: FilterOptions<T>
+): FilterGroup[] {
+    groups.forEach(({ values, name }) => {
+        const customSort = filterConfig.find(({ name: configName }) => name === configName)?.sort;
+
+        if (customSort) {
+            customSort(values);
+        } else {
```
Comment (Gustav-Eikaas) [inline src/packages/Filter/functions/generateFilterValues.ts:78]: Fix sorting

**Review by jaysencpp — APPROVED** [review]

**Pull request merged**